src/Common/TransformEndianness.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <base/Decimal_fwd.h>
+#include <base/extended_types.h>
+
+#include <utility>
+
+namespace DB
+{
+template <std::endian endian, typename T>
+requires std::is_integral_v<T>
+inline void transformEndianness(T & value)
+{
+    if constexpr (endian != std::endian::native)
+        value = std::byteswap(value);
+}
+
+template <std::endian endian, typename T>
+requires is_big_int_v<T>
+inline void transformEndianness(T & x)
+{
+    if constexpr (std::endian::native != endian)
+    {
+        auto & items = x.items;
+        std::transform(std::begin(items), std::end(items), std::begin(items), [](auto & item) { return std::byteswap(item); });
+        std::reverse(std::begin(items), std::end(items));
+    }
+}
+
+template <std::endian endian, typename T>
+requires is_decimal<T>
+inline void transformEndianness(T & x)
+{
+    if constexpr (std::endian::native != endian)
```

Comment:
(minor: could omit l. 34 - we also have no `if constexpr` in the `is_scoped_enum` and `std::pair` handlers)

Reply:
Well spotted, thanks!